Gems/AtomLyIntegration/CommonFeatures/Code/Source/SharedPreview/SharedThumbnailRenderer.cpp

```diff
@@ -51,64 +54,124 @@ namespace AZ
+                        AtomToolsFramework::GetSettingsValue<AZStd::string>(LightingAssetPathSetting, DefaultLightingPresetPath)));
+                }
+
+                return thumbnailConfig;
+            }
+
+            const auto& path = AZ::RPI::AssetUtils::GetSourcePathByAssetId(assetInfo.m_assetId);
+            if (assetInfo.m_assetType == RPI::AnyAsset::RTTI_Type() && AZ::StringFunc::EndsWith(path.c_str(), ".lightingpreset.azasset"))
             {
                 static constexpr const char* ModelAssetPathSetting =
                     "/O3DE/Atom/CommonFeature/SharedPreview/LightingAssetType/ModelAssetPath";
                 static constexpr const char* MaterialAssetPathSetting =
                     "/O3DE/Atom/CommonFeature/SharedPreview/LightingAssetType/MaterialAssetPath";
 
-                thumbnailConfig.m_modelId = SharedPreviewUtils::GetAssetIdForProductPath(
-                    AtomToolsFramework::GetSettingsValue<AZStd::string>(ModelAssetPathSetting, DefaultModelPath));
-                thumbnailConfig.m_materialId = SharedPreviewUtils::GetAssetIdForProductPath(
-                    AtomToolsFramework::GetSettingsValue<AZStd::string>(MaterialAssetPathSetting, "materials/reflectionprobe/reflectionprobevisualization.azmaterial"));
-                thumbnailConfig.m_lightingId = assetInfo.m_assetId;
+                ThumbnailConfig thumbnailConfig;
+                thumbnailConfig.m_modelAsset.Create(SharedPreviewUtils::GetAssetIdForProductPath(
+                    AtomToolsFramework::GetSettingsValue<AZStd::string>(ModelAssetPathSetting, DefaultModelPath)));
+                thumbnailConfig.m_materialAsset.Create(
+                    SharedPreviewUtils::GetAssetIdForProductPath(AtomToolsFramework::GetSettingsValue<AZStd::string>(
+                        MaterialAssetPathSetting, "materials/reflectionprobe/reflectionprobevisualization.azmaterial")));
+                thumbnailConfig.m_lightingAsset.Create(assetInfo.m_assetId);
+                return thumbnailConfig;
             }
 
-            return thumbnailConfig;
+            if (assetInfo.m_assetType == RPI::AnyAsset::RTTI_Type() && AZ::StringFunc::EndsWith(path.c_str(), ".modelpreset.azasset"))
+            {
+                static constexpr const char* MaterialAssetPathSetting =
+                    "/O3DE/Atom/CommonFeature/SharedPreview/ModelAssetType/MaterialAssetPath";
+                static constexpr const char* LightingAssetPathSetting =
+                    "/O3DE/Atom/CommonFeature/SharedPreview/ModelAssetType/LightingAssetPath";
+
+                // Model preset assets are relatively small JSON files containing a reference to a model asset and possibly other
+                // parameters. The preset must be loaded to get the model asset ID. Then the preview will be rendered like any other model.
+                AZ::Data::Asset<AZ::RPI::AnyAsset> asset = AZ::RPI::AssetUtils::LoadAssetById<AZ::RPI::AnyAsset>(assetInfo.m_assetId);
```

Comment:
This is a blocking load, is there any chance of a BlockUntilLoadComplete deadlock here?

Reply:
I wonder if this should work more like the lighting preset where you store the preset asset in the ThumbnailConfig instead of the ModelAsset

Reply:
I was trying to do a quick load to resolve the model asset then treat it the same as the normal model asset after that. If there is a concern over a deadlock then I could bypass the asset system and read the json object. If we are not adding other fields to the model preset we should consider getting rid of it later.